### PR TITLE
Adjusted Required Scenario Distribution to More Evenly Award CVP; Reduced Scenario Sizes for House & Integrated Commands

### LINKED ISSUE #5221: [RFE] Equalizing mission VPs
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the RFE to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] Does this conform with our Content Policy linked above?
- [X] I've searched the GitHub tracker and haven't found a similar feature request listed

### RFE Details *

This is an improvement to an existing feature

### Brief Description *

Currently certain mission types - like Relief Duty or Garrison - have significantly fewer VP-critical scenarios than others and it's possible to go an entire mission with few or even zero VPs being able to be awarded. Meanwhile other types will award VPs like candy.

As the new XP defaults specifically award significant bonuses for completing a mission with 3+ VPs, this makes certain mission types undesirable while others much more rewarding. (More so, the difference is invisible to the average player as there's no indication which ones are superior.) There should probably be some form of check to ensure that any given mission at least provides the player the _opportunity_ to reach the 3 VP threshold. 

### MekHQ Suite Version *

v0.50.0

### Attach Files

_No response_

### Final Checklist

- [X] I've checked to make sure that this RFE has not already been filed
- [X] I am requesting implementation of only one relevant set of features in this ticket for clarity and focus

### PR DESCRIPTION
**Pull request by IllianiBird**
Updated the 'required scenario' logic used by StratCon to more evenly distribute 'required' status. Previously, Independent contracts were unable to accumulate CVP, as none of their scenarios were considered 'required'.

I also went ahead and updated the frequency in which attached units would be applied to scenarios to try and bring scenario size down a little. Already we see quite large scenarios occurring and while some players enjoy that, I don't think it's good to have every scenario be a company-level engagement.

Now a scenario's 'required' status is determined as so:
- INTEGRATED, HOUSE: all scenarios are required. 25% of scenarios feature attached allies.
- LIAISON: 25% of scenarios are required. All required scenarios feature an attached ally (the forward observer).
- INDEPENDENT: 25% of scenarios are required. No attached allies.

Finally, I moved Cadre Duty's trainees away from the required scenario logic. That ensures Trainees can appear in addition to attached allies. Meaning that on a Liaison Cadre Duty you may find you're deploying alongside both Trainees and a Forward Observer (probably a political officer, let's be fair).

Fixes #5221

### CODE REVIEW

**Comment by BLR-IIC** [comment]
A thought about the company-size and larger engagements - I really like how the automatic resolution system, especially the abstract combat system, makes these kinds of battles trivial.  But I'm not sure about how to allow the player to determine the engagement size, especially in an immersive way.

**Comment by IllianiCBT** [comment]
Check the Combat Team documentation, when 50.02 drops. That tells you how :)

**Comment by HammerGS** [comment]
@IllianiCBT Conflict and Alert

File: MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java [hunk 1]
```diff
@@ -606,6 +610,97 @@
...
+        boolean isResupply = scenario.getBackingScenario().getStratConScenarioType().isResupply();
+
+        if (isResupply) {
+            scenario.setRequiredScenario(false);
+            return;
+        }
+
+        if (template == null || !template.getStratConScenarioType().isResupply()) {
+            ContractCommandRights commandRights = contract.getCommandRights();
+            switch (commandRights) {
```
[the remark this answers is not in the document]
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java:682]: Yes it does...

File: MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java [hunk 2]
```diff
@@ -607,6 +611,97 @@
...
+        boolean isResupply = scenario.getBackingScenario().getStratConScenarioType().isResupply();
+
+        if (isResupply) {
+            scenario.setTurningPoint(false);
+            return;
+        }
+
+        if (template == null || !template.getStratConScenarioType().isResupply()) {
+            ContractCommandRights commandRights = contract.getCommandRights();
+            switch (commandRights) {
```
[the remark this answers is not in the document]
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java:683]: It's literally there

**Pull request merged**